shell/server/proxy.js

```diff
@@ -380,7 +381,23 @@ Meteor.methods({
       }
     }
 
-    if (this.userId && !incognito) {
+    if (this.userId && standalone) {
+      if (!SandstormPermissions.mayOpenGrain(globalDb, { token: apiToken })) {
```

Comment:
This block appears to be an exact copy of the anonymous case, below. Please just adjust the if() statement below so that these cases take the same code path, rather than copy/pasting a bunch of code.